core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -301,6 +296,22 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException, UserError {
         return target;
     }
 
+    private Path stagingDirectory(Path pluginsDir) throws IOException {
+        try {
+            Set<PosixFilePermission> perms = new HashSet<>();
+            perms.add(PosixFilePermission.OWNER_EXECUTE);
+            perms.add(PosixFilePermission.OWNER_READ);
+            perms.add(PosixFilePermission.OWNER_WRITE);
+            perms.add(PosixFilePermission.GROUP_READ);
+            perms.add(PosixFilePermission.GROUP_EXECUTE);
+            perms.add(PosixFilePermission.OTHERS_READ);
+            perms.add(PosixFilePermission.OTHERS_EXECUTE);
+            return Files.createTempDirectory(pluginsDir, ".installing-", PosixFilePermissions.asFileAttribute(perms));
+        } catch (IllegalArgumentException | UnsupportedOperationException e) {
+            return Files.createTempDirectory(pluginsDir, ".installing-");
```

Comment:
@rmuir Unfortunately, the `IllegalArgumentException` catch is needed because JimFS violates the javadocs for `FileSystemProvider#createDirectory`; instead of throwing an `UnsupportedOperationException` when an attempt is made to set POSIX attributes on a file on a filesystem that does not support POSIX attributes. I'll hack the catch though to only catch it when it's coming from jimfs. I'll also submit an upstream pull request to get this corrected, but for now I think this is what we have to do?